python/tchannel/rw.py

```diff
@@ -437,9 +437,13 @@ def read(self, stream):
                 if attr != skip:
                     kwargs[attr] = value
         except ReadException as e:
-            raise ReadException(
-                "Failed to read %s: %s" % (self._cls, e.message)
-            )
+            """ capture the read exception msg here but
```

Comment:
Don't catch this here. Instead of breaking all things that use this read writer, just override the behavior for specific instances.